Pearson-Advance/frontend-app-institution-portal
# feat: availability tab with calendar

# Description

This PR adds a new tab `Availability` in instructor detail page along with a calendar with its events.

This PR resolves [PADV-1760](https://agile-jira.pearson.com/browse/PADV-1760)

## Change log
- Renamed test folder in instructors detail page and data folder to use double `_` instead a single `_`
- Re factorization in `useEffect`, spitted responsibilities and added clean methods for `classes` and `events`
- Added flag `enable_instructor_calendar` to show the calendar along with its events
- Updated unit test

### Visual results
#### Before
![before](https://github.com/user-attachments/assets/c2a19f2d-fcd9-4929-95ed-edbcae446cfd)

#### After
![after](https://github.com/user-attachments/assets/1e8a7083-688b-4440-ad98-3fe802a34995)


### How to test
- Configure the flag `enable_instructor_calendar` in the site admin 18000 or override the content  of the request `http://localhost:18000/api/mfe_config/` to have the flag in `true`

- Go to '/instructors/:instructorUsername' if `enable_instructor_calendar` is `true` you should be able to see the calendar along with the `events/` request, if the flag is `false` you shouldn't see any of both.
